wasm/wasm_source/src/vp_masp.rs

```diff
@@ -55,7 +97,32 @@ fn validate_tx(
         transparent_tx_pool += shielded_tx.value_balance.clone();
 
         // Handle shielding/transparent input
+        // The following boundary conditions must be satisfied
+        // 1. One transparent input
+        // 2. Zero transparent output
+        // 3. the transparent transaction value pool's amount must equal the
+        // containing wrapper transaction's fee amount
         if transfer.source != masp() {
```

Comment:
Where do we check for the for the following condition?: If the source address is the MASP validity predicate, then no transparent inputs are permitted in the shielded transaction.

It seems to me that if the source and target are MASP, then both the conditional on line 105 and line 146 are skipped. This would allow for these sorts of transactions to have unlimited transparent inputs and outputs.